webapp/src/ts/components/filters/date-filter/date-filter.component.ts

```diff
@@ -93,7 +93,7 @@ export class DateFilterComponent implements OnInit, OnDestroy, AbstractFilter, A
       if (this.isRange) {
         return;
       }
-      let date = this.isStartDate ? this.dateRange.from : this.dateRange.to;
+      let date: moment.Moment | undefined = this.isStartDate ? this.dateRange.from : this.dateRange.to;
       if (!date) {
         date = moment();
       }
```

Comment:
To me this is potentially worse than it was, and it was pretty bad before! :laughing: 

With the typing now you're letting the IDE know it might be undefined, but the next line ensures that it's NOT undefined. I'm not sure my IDE is smart enough to figure that out.

I think a better solution would be to extract a function which returns the date which does the ternary logic and the null check and returns a moment.Moment which can never be undefined. This also means it can be a `const` (I think).

Reply:
Ah I forgot to come and roll back these changes after updating the `dateRange` types above.  AFICT this `moment.Moment | undefined` typing is wrong since these can actually have number values as well...

Either way, I don't actually need to specify the type here for `date` at all.  I 100% agree that this code is not great and could be refactored to be much better, but when possible for this PR, I think it will be best to avoid making non-type changes.